pyqtgraph/SignalProxy.py

```diff
@@ -76,7 +76,7 @@ def flush(self):
             return False
         args, self.args = self.args, None
         self.timer.stop()
-        self.lastFlushTime = time()
+        self.lastFlushTime = perf_counter()
         self.sigDelayed.emit(args)
         return True
 
```

Comment:
I guess this needs perf_counter (and not just time.time() ) to allow precise rate limiting at short intervals?
 
Unrelated to this PR:
Below this is a the disconnect function with a comment regarding a workaround for PySide on Python 2.
Do you know that that is about and if it can be removed or simplified now?

Reply:
Fairly sure it can be simplified/removed now... although that should probably be its own commit in case we have to git-bisect our way back.  PySide1 has almost no resemblance to PySide2 of today (besides the name)

Reply:
looking a bit more on it, this can be simplified a great deal now with the use of [`weakref.finalize`](https://docs.python.org/3/library/weakref.html#weakref.finalize) which is new in python 3.4

Reply:
actually while I just tried this, I realized there is a gotcha with weakref.finalize, you cannot reference the object being finalized within the method being called...which is a bit problematic due to how we handle the signals... might be worth revisiting more later, but going to file this under "if it's stupid, and it works, it's not stupid".

Reply:
If your reading of the comment is "This implementation was chosen because it also works on Pyside" then I agree.
At first glance the current code looked reasonable enough. It also did not look like this wouldn't typically be part of anything time-critical. So, seems good not to touch it unless you see a nice way to make it elegant!

Reply:
I've been thinking of drafting a PR to try and remove some of these Qt4 work-arounds/band-aids one commit at a time (for git bisect purposes)... I'll be sure to try and clean this up then!